pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java

```diff
@@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
   }
 
   public CopyIn copyIn(String sql) throws SQLException {
-    CopyOperation op = null;
-    try {
-      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    if (op == null || op instanceof CopyIn) {
```

Comment:
Faster to execute

Dave Cramer

On 5 March 2016 at 09:03, Vladimir Sitnikov <notifications@github.com>
wrote:

> In pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java
> <https://github.com/pgjdbc/pgjdbc/pull/527#discussion_r55121473>:
>
> > @@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
> >    }
> >
> >    public CopyIn copyIn(String sql) throws SQLException {
> > -    CopyOperation op = null;
> > -    try {
> > -      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
> > +    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
> > +    if (op == null || op instanceof CopyIn) {
>
> Do you mean "faster to read" or "faster to execute"?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/pgjdbc/pgjdbc/pull/527/files#r55121473>.
>